src/librustc_resolve/resolve_imports.rs

```diff
@@ -783,7 +783,7 @@ impl<'a, 'b:'a> ImportResolver<'a, 'b> {
             };
 
             if binding.vis == ty::Visibility::Public &&
-               (binding.is_import() || binding.is_extern_crate()) {
+               (binding.is_import() || binding.is_macro_def()) {
```

Comment:
Removing `binding.is_extern_crate()` is a no-op (implied by `binding.is_import()`), adding `binding.is_macro_def()` allows the binding to be accessed from external crates.